app/controllers/books_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class BooksController < ApplicationController
+  before_action :set_book, only: %i[show edit update destroy]
+
+  def index
+    @books = Book.page params[:page]
+  end
+
+  def new
+    @book = Book.new
+  end
+
+  def show; end
+
+  def edit; end
```

Comment:
don't define empty methods in case of `before_action`

Reply:
done